src/Storages/MergeTree/IExecutableTask.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+
+#include <common/shared_ptr_helper.h>
+#include <Interpreters/StorageID.h>
+
+namespace DB
+{
+
+class IExecutableTask
+{
+public:
+    virtual bool execute() = 0;
+    virtual void onCompleted() = 0;
+    virtual StorageID getStorageID() = 0;
+    virtual ~IExecutableTask() = default;
+};
+
+using ExecutableTaskPtr = std::shared_ptr<IExecutableTask>;
+
+
+class LambdaAdapter : public shared_ptr_helper<LambdaAdapter>, public IExecutableTask
```

Comment:
Renamed to `ExecutableLambdaAdapter for this name not to be so generic and added comments. Yes, we need `IExecutableTask` because this class implements this interface.